scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -498,22 +499,23 @@ def process(self, pipeline, done, message, lock, results):
                 done.put(self.instance)
                 self.report_out(results)
 
-            if self.options.runtime_artifact_cleanup and not self.options.coverage and self.instance.status == "passed":
+            if (self.options.runtime_artifact_cleanup == 'passing-tests' and not self.options.coverage and self.instance.status == "passed")\
```

Comment:
this conditional is becoming unreadable and too long, please reformat or simplify/split the logic.